smartcontract/states/contractstate.go

```diff
@@ -0,0 +1,114 @@
+package states
+
+import (
+	"io"
+	. "DNA/errors"
+	"DNA/core/code"
+	"bytes"
+	"DNA/common/serialization"
+	"DNA/smartcontract/types"
+	"DNA/common"
+)
+
+type ContractState struct {
+	Code *code.FunctionCode
+	Name string
+	Version string
+	Author string
+	Email string
+	Description string
+	Language types.LangType
+	ProgramHash common.Uint160
+	*StateBase
```

Comment:
use object instead of pointer

Reply:
why?

Reply:
for compact memory layout and performance since StateBase  just consists 1byte , then what is the point to use another 8bit pointer ?

Reply:
this is a question